src/dotnet/CommandDirectoryContext.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.IO;
+
+namespace Microsoft.DotNet.Cli
+{
+    internal static class CommandDirectoryContext
+    {
+        [ThreadStatic]
+        private static string _currentDirectoryOverwrite;
+
+        /// <summary>
+        /// Expands a path similar to Path.GetFullPath() but allows unit tests a hook to inject an overwrite to the
+        /// current working directory.
+        /// </summary>
+        /// <param name="path">A realtive or absolute path specifier</param>
+        /// <returns>The full path to the target</returns>
+        public static string ExpandPath(string path)
+            => _currentDirectoryOverwrite is string basePath
```

Comment:
Is the cast here necessary rather than a null check?

Reply:
I should look at the generated code but wanted to avoid an extra TLS read but since there are no perf considerations I'll do it

Reply:
If the extra read is a concern, then I think this would be better as normal function implementation that declared a local with a small comment explaining why the local is being assigned.